src/map_calc.cpp

```diff
@@ -71,6 +72,28 @@ double eval_grp_hinge(const subview_col<uword> &districts, int distr,
     return std::sqrt(std::max(0.0, target - frac));
 }
 
+/*
+ * Compute the new, hinge group penalty for district `distr`
+ */
+double eval_grp_inv_hinge(const subview_col<uword> &districts, int distr,
```

Comment:
this is duplicative & so will add compilation time & make it hard to maintain. 

we just need to pass the arguments to `eval_grp_hinge` but with the order reversed.